tigershark/facade/__init__.py

```diff
@@ -295,6 +297,25 @@ def loops(self, theClass, anX12Message, *args, **kwargs):
         return [theClass(loop, *args, **kwargs) for loop in
                 anX12Message.descendant("loop", theClass.loopName)]
 
+    def to_dict(self):
+        return Facade._to_python_dict(self)
+
+    def to_json(self, **kwargs):
+        return json.dumps(self.to_dict(), **kwargs)
+
+    @staticmethod
+    def _to_python_dict(instance):
+        properties = [x for x in inspect.getmembers(instance) if not x[0].startswith("_")]
+        record = dict()
+        for k, v in properties:
+            if isinstance(v, list):
+                record[k] = [Facade._to_python_dict(x) for x in v]
+            elif type(v) in [int, float, str, bool]:
+                record[k] = v
+            elif isinstance(v, X12LoopBridge) or isinstance(v, X12SegmentBridge) or isinstance(v, Facade):
+                record[k] = Facade._to_python_dict(v)
```

Comment:
What do you think about making `X12LoopBridge` and `X12SegmentBridge` implement this as well? Then we can just test for `hasattr('_to_python_dict')` and call it.